fp-background.js

```diff
@@ -0,0 +1,450 @@
+  await messenger.Utilities.initFolderPaneOptionsPopup();
+
+  /**/
+  await messenger.LegacyMenu.add(id, {
+    "id": "FolderPaneSwitcher-forward-arrow-button",
+    "type": "toolbarButton",
+    "reference": "folderPaneOptionsButton",
+    "position": "before",
+    "label": "",
+    //  "accesskey" : "B",
+    "image": "content/right-arrow.png",
+    "tooltip": "Next View"
+  });
+
+
+
+  await messenger.LegacyMenu.add(id, {
+    "id": "FolderPaneSwitcher-back-arrow-button",
+    "type": "toolbarButton",
+    "reference": "FolderPaneSwitcher-forward-arrow-button",
+    "position": "before",
+    "label": "",
+    //  "accesskey" : "B",
+    "image": "content/left-arrow.png",
+    "tooltip": "Previous View"
+  });
+
+
+};
+
+
+
+messenger.windows.onRemoved.addListener(async (windowId) => { await messenger.Utilities.unregisterListener("all", "folderPaneHeader", windowId); })
+
+messenger.LegacyMenu.onCommand.addListener(async (windowsId, id) => {
+  if (id == "FolderPaneSwitcher-forward-arrow-button") {
+    //console.log("forward clicked");
+    FolderPaneSwitcher.goForwardView();
+    return;
+  };
+
+  if (id == "FolderPaneSwitcher-back-arrow-button") {
+    //console.log("back clicked");
+    FolderPaneSwitcher.goBackView();
+    return;
+  };
+
+
+});
+
+var FolderPaneSwitcher = {
+
+  cachedView: null,
+
+  selectedFolder: null,  //not in use??
+
+  timer: 0,
+
+  watchTimer: 0,
+
+  logger: console,
+
+
+  setSingleMode: async function (modeName) {
+    let activeModes = await messenger.Utilities.getActiveViewModes();
+    let currModes = activeModes.slice();
+    if (!activeModes.includes(modeName)) await messenger.Utilities.toggleActiveViewMode(modeName);
+
+    for (viewName of currModes) {
+      if (viewName != modeName) await messenger.Utilities.toggleActiveViewMode(viewName);
+    }
+  },
+
+  goForwardView: async function (event) {
+
+    let activeModes = await messenger.Utilities.getActiveViewModes();
+    let arrowViews = await messenger.storage.local.get("arrowViews");
+    selectedViews = arrowViews.arrowViews;
+    //console.log("selectedViews", selectedViews);
+
+    var currentView = activeModes[activeModes.length - 1];
+    let currInd = selectedViews.findIndex((name) => name == currentView);
+    currInd = (currInd + 1) % selectedViews.length;
+    FolderPaneSwitcher.setSingleMode(selectedViews[currInd]);
+  },
+
+  goBackView: async function () {
+
+    let activeModes = await messenger.Utilities.getActiveViewModes();
+    let arrowViews = await messenger.storage.local.get("arrowViews");
+    selectedViews = arrowViews.arrowViews;
+    //console.log("selectedViews", selectedViews);
+
+    var currentView = activeModes[activeModes.length - 1];
+    let currInd = selectedViews.findIndex((name) => name == currentView);
+
+    currInd = (currInd + selectedViews.length - 1) % selectedViews.length;
+    FolderPaneSwitcher.setSingleMode(selectedViews[currInd]);
+  },
```

Comment:
which delay do you mean?
There is a delay upon dragging an email into the folderpane header, which then switches from current view to all view. That is implemented.

On the other hand, I think Jonathan asked his previous coworker why he set that delay high instead of setting to very short, soI am not sure how relevant that actually is. If i drag the email there,i express my intention for the view to be adjusted, so why do it after a delay?


So which delay do you refer to?

Reply:
I think this is ok as per our discussion